### PR TITLE
fix error in priced? check when product does not have pricing_details

### PR DESCRIPTION
**Pull request by tlclink**

### CODE REVIEW

**Review by ZHENG-JING-JZ — APPROVED** [review]

**Comment by tlclink** [comment]
The CI errors are all bundler errors so I'm guessing its a CI configuration problem
@HellRok who manages those configurations?

**Comment by HellRok** [comment]
@tlclink The configurations are commited into the repository (look into .travis.yml)

Very strange it failed on 2.3 but passed on all other versions (including 2.2), might be worth changing the script line to be `script: gem install bundler && bundle && bundle exec rake spec`

**Comment by tlclink** [comment]
@HellRok sadly that didn't make any difference :(

**Comment by jmortlock** [comment]
There are a few ways to fix this.
Update our Rails4 GemSpec to match the bundler requirements from upstream rails
```
s.add_dependency 'bundler', '>= 1.3.0', '< 2.0'
```
Or can always enforce bundler < 2 by pre-installing
```
before_install:
  gem install bundler -v "~>1.0"
```
```

File: lib/quick_travel/product_configuration.rb [hunk 1]
```diff
@@ -31,7 +31,7 @@ def available?
     end
 
     def priced?
-      @product.pricing_details.present?
+      @product.respond_to?(:pricing_details) && @product.pricing_details.present?
```
Comment (jmortlock) [inline lib/quick_travel/product_configuration.rb]: Can do
```
@product.try(:pricing_details).present?
```
And update the unit tests to test this scenario

**Comment by tlclink** [comment]
@jmortlock 
> There are a few ways to fix this.
> Update our Rails4 GemSpec to match the bundler requirements from upstream rails
> 
> ```
> s.add_dependency 'bundler', '>= 1.3.0', '< 2.0'
> ```
> Or can always enforce bundler < 2 by pre-installing
> 
> ```
> before_install:
>   gem install bundler -v "~>1.0"
> ```

Did you mean to apply the `add_dependency` change in the `quicktravel_client.gemspec` or a different file?

**Comment by jmortlock** [comment]
gemfiles/rails4.gemfile

**Comment by tlclink** [comment]
@jmortlock unfortunatley neither of your suggestions appear to work :(
unless I didn't apply the suggestions correctly

**Comment by jmortlock** [comment]
Nice one, 6th times a charm

**Review by jmortlock — APPROVED** [review]

**Pull request merged**